src/main/java/com/simibubi/create/content/trains/entity/Train.java

```diff
@@ -150,6 +150,9 @@ public class Train {
 	int ticksSinceLastMailTransfer;
 	double[] stress;
 
+
+	private CollisionCache collisionCache;
```

Comment:
Why rely on both nullability and an invalid flag? Is there a way to only use one or the other?